### PR TITLE
[vk-bootstrap] set missing IMPORTED_LOCATION

### LINKED ISSUE #22067: [vk-bootstrap] Missing import target when building on Linux
**Host Environment**
- OS: x86_64 Linux 5.8.0-43-generic
- Compiler: g++-11 (Ubuntu 11.1.0-1ubuntu1~20.04)

**To Reproduce**
This happens when trying to configure a cmake project depending on `vk-bootstrap` while on Linux.
```cmake
find_package(vk-bootstrap CONFIG REQUIRED)
target_link_libraries(carbon PUBLIC vk-bootstrap::vk-bootstrap)
```

`./vcpkg install vk-bootstrap`
`mkdir build && cd build`
`cmake ..`

**Failure logs** 
```
CMake Error in src/carbon/CMakeLists.txt:
  IMPORTED_LOCATION not set for imported target "vk-bootstrap::vk-bootstrap"
  configuration "Debug".


CMake Error in src/carbon/CMakeLists.txt:
  IMPORTED_LOCATION not set for imported target "vk-bootstrap::vk-bootstrap"
  configuration "Debug".


CMake Error in src/carbon/CMakeLists.txt:
  IMPORTED_LOCATION not set for imported target "vk-bootstrap::vk-bootstrap"
  configuration "Debug".
```

**Additional context**
I know I made the `vk-bootstrap` port but it seems like it only worked on Windows, even though it's the same CMake project I'm testing right now. I'm no expert in CMake, but this seems like it's a pretty simple fix that I am just missing.

### PR DESCRIPTION
**Pull request by FrankXie05**
Fix #22067

Note: Set the parameter `IMPORTED_LOCATION` on the usage of port `vk-bootstrap`.
Error log: 
````
CMake Warning (dev) in CMakeLists.txt:
  Policy CMP0111 is not set: An imported target missing its location property
  fails during generation.  Run "cmake --help-policy CMP0111" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  IMPORTED_LOCATION not set for imported target "vk-bootstrap::vk-bootstrap".
This warning is for project developers.  Use -Wno-dev to suppress it.
````

### CODE REVIEW

File: ports/vk-bootstrap/vk-bootstrap-config.cmake [hunk 1]
```diff
@@ -2,10 +2,12 @@ get_filename_component(_IMPORT_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
 get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
 get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
 
+set(vk-bootstrap_lib  "${_IMPORT_PREFIX}/lib/vk-bootstrap.lib")
 add_library(vk-bootstrap::vk-bootstrap SHARED IMPORTED)
 set_target_properties(vk-bootstrap::vk-bootstrap PROPERTIES
     INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include;${_IMPORT_PREFIX}/include"
 )
+set_target_properties(vk-bootstrap::vk-bootstrap PROPERTIES IMPORTED_LOCATION "${vk-bootstrap_lib}")
```
Comment (dg0yt) [inline ports/vk-bootstrap/vk-bootstrap-config.cmake:10]: - For a `SHARED` windows lib, I would exect `IMPORTED_IMPLIB`.
- Can't the property be added to the previous command?
- Do you need the `vk-bootstrap_lib` variable?
- Is it really the same for release and debug? (I don't know.)

Comment (FrankXie05) [reply ports/vk-bootstrap/vk-bootstrap-config.cmake:10]: @dg0yt  
1. For `SHARED` libraries, `IMPORTED_IMPLIB` is better. :)
2. For port `vk-bootstrap`, I see that they are improving the usage of exporting cmake, I will ask if they will release a new version. https://github.com/charles-lunarg/vk-bootstrap/blob/b01b2b1d8ef3fb1d784f1f53d716995855d89ea2/CMakeLists.txt#L79
3. I should probably update this port and use the upstream provided targets, and drop the current `*-config.cmake`. 🤔

**Comment by FrankXie05** [comment]
Turn this PR into a draft.

File: ports/vk-bootstrap/vk-bootstrap-targets-debug.cmake [hunk 2]
```diff
@@ -7,6 +7,7 @@ set(CMAKE_IMPORT_FILE_VERSION 1)
 set_property(TARGET vk-bootstrap::vk-bootstrap APPEND PROPERTY IMPORTED_CONFIGURATIONS DEBUG)
 set_target_properties(vk-bootstrap::vk-bootstrap PROPERTIES
     IMPORTED_IMPLIB_DEBUG "${_IMPORT_PREFIX}/debug/lib/vk-bootstrap.lib"
+    IMPORTED_LOCATION "${_IMPORT_PREFIX}/debug/lib/vk-bootstrap.lib"
```
Comment (JackBoosY) [inline ports/vk-bootstrap/vk-bootstrap-targets-debug.cmake:10]: The origin cmake configurations are all wrong. 

`vk-bootstrap` should supports all the triplet, but there is only Windows.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/vk-bootstrap/vk-bootstrap-targets-debug.cmake (hunk 2 above)
Comment (FrankXie05) [reply ports/vk-bootstrap/vk-bootstrap-targets-debug.cmake:10]: I will recheck the cmake configuration of the port

**Comment by JonLiu1993** [comment]
Draft this pr because it’s been a long time since any update has been made

**Comment by JonLiu1993** [comment]
Closing this PR since it seems that no progress is being made. Please ping us to reopen if work is still being done.

**Pull request closed without merging**